### ISSUE TITLE
How do we encourage and possibly enforce accessibility at a fundamental level

### ISSUE THREAD

**Issue by fabiancook** [opennetwork/vnode#14]
Given the idea of dynamic fragments we can start to shape the model utilised by the ecosystem of iterable

As more of the ecosystem materializes we will get further away from being able to tackle accessibility directly without causing either excuses, confusion, lack of back maintenance, or rushed schedules. 

At the fundamental level we know the entire view state, given it is represented, this allows us to statically analyse the entire state at any given time, we can run audits at any point, or collect the entire session, which includes any automated testing. This allows us to cycle through every function available, which can then be also sandboxed into accessibility zones. 

If a region isn't accessible to a user, it is not useful to a user. 

This includes the following possible reasons:

- the consumer is a machine and only understands a specific type of source reference (e.g. scalar flagged with a data type)
- the consumer has no device to materialize the state
- the consumer has a sensitivity (e.g. flashing visual, or loud sound), or has disabled the given state/media type/content
- the consumer has not got permission or access to the state 

If we have these fragments defined at a fundamental level, along with possible additional requirements, then a consumer can direct how they want to consume the state. 

An audit can be created by a private or public trusted third party to verify the states capabilities if required, for example if the consumer has requested closed captions for English, then closed captions for English must be available, and they must be accurate to the consumers requirements. 

This can be extended to data apis, to enforce document state structure when a consumers data is to be stored on behalf. 

This approach is the forced approach and may not give consumers the best experience, forcing the consumer to then rethink their requirements. 

But, if the results are public for a view state, then it would encourage full coverage of accessing the view state. 

The requirements described here doesn't cover everything accessibility, but it does indicate the direction I believe could be a possible way forward. 

This would still be a function that can be disabled, disregarded, and put to the side, but it could also be a useful tool for consumers, which extends to developers who want to ensure that a consumer has the best experience that they possibly can.
